src/submodules/AtomsCalculatorsTesting.jl

```diff
@@ -0,0 +1,116 @@
+        f_cpu_array = Array(f)  # Allow GPU output
+        @test dimension(f_cpu_array[1][1]) == dimension(u"N")
+        @test length(f_cpu_array[1]) == (length ∘ position)(sys,1)
+        f_nonallocating = zeros(ftype, length(sys))
+        AtomsCalculators.forces!(f_nonallocating, sys, calculator; kwargs...)
+        @test all( f_nonallocating .≈ f  )
+        AtomsCalculators.forces!(f_nonallocating, sys, calculator; dummy_kword659254=1, kwargs...)
+        @test all( f_nonallocating .≈ f  )
+        fc = AtomsCalculators.calculate(AtomsCalculators.Forces(), sys, calculator; kwargs...)
+        @test all( f .≈ fc )
+    end
+end
+
+
+"""
+    test_potential_energy(sys, calculator; kwargs...)
+
+Test your calculator for AtomsCalculators interface. Passing test means that your
+potential energy calculation.
+
+To use this function create an `AtomsBase` system `sys` and a `calculator` for your
+own calculator. Test function will then call the interface and performs checks
+for the output and checks that random keywords are accepted in input. 
+
+`kwargs` can be passed to the `calculator` for tuning during testing.
+
+The calculator is expected to work without kwargs.
+"""
+function test_potential_energy(sys, calculator; kwargs...)
+    @testset "Test potential_energy for $(typeof(calculator))" begin
+        e = AtomsCalculators.potential_energy(sys, calculator; kwargs...)
+        @test typeof(e) <: Number
+        @test dimension(e) == dimension(u"J")
+        e2 = AtomsCalculators.potential_energy(sys, calculator; dummy_kword6594254=1, kwargs...)
+        @test e ≈ e2
+        ec = AtomsCalculators.calculate(AtomsCalculators.Energy(), sys, calculator; kwargs...)
+        @test e ≈ ec
+    end
+end
+
+
+"""
+    test_virial(sys, calculator; kwargs...)
+
+Test your calculator for AtomsCalculators interface. Passing test means that your
+virial calculation.
+
+To use this function create an `AtomsBase` system `sys` and a `calculator` for your
+own calculator. Test function will then call the interface and performs checks
+for the output and checks that random keywords are accepted in input. 
+
+`kwargs` can be passed to the `calculator` for tuning during testing.
+
+The calculator is expected to work without kwargs.
+"""
+function test_virial(sys, calculator; kwargs...)
+    @testset "Test virial for $(typeof(calculator))" begin
+        v = AtomsCalculators.virial(sys, calculator; kwargs...)
+        @test typeof(v) <: AbstractMatrix
+        @test eltype(v) <: Number
+        v_cpu_array = Array(v) # Allow GPU arrays
+        @test dimension(v_cpu_array[1,1]) == dimension(u"J")
+        l = (length ∘ position)(sys,1) 
+        @test size(v) == (l,l) # allow different dimensions than 3
+        v2 = AtomsCalculators.virial(sys, calculator; dummy_kword6594254=1, kwargs...)
+        @test all( v .≈ v2 )
+        vc = AtomsCalculators.calculate(AtomsCalculators.Virial(), sys, calculator; kwargs...)
+        @test all( v .≈ vc )
+    end
+end
+
+
```

Comment:
Test calculate ?

Reply:
Calculate method is tested within the corresponding energy, forces or virial tests. So, I did not implement it separately.